app/src/main/java/com/cpr3663/cpr_scouting_app/Match.java

```diff
@@ -328,15 +344,14 @@ public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMen
                 is_start_of_seq = true;
             } else {
                 events_al = Globals.EventList.getNextEvents(eventPrevious);
-                if (events_al == null) {
+                if ((events_al == null) || events_al.isEmpty()) {
                     events_al = Globals.EventList.getEventsForPhase(matchPhase);
                     is_start_of_seq = true;
                 }
             }
-            events = new String[events_al.size()];
-            events = events_al.toArray(events);
+
             // Add all the events
-            for (String event : events) {
+            for (String event : events_al) {
```

Comment:
Now that we fixed it and it isn't converted to an array you could change the name to events instead of events_al.